solr/core/src/test/org/apache/solr/cloud/TestConfigSetsAPI.java

```diff
@@ -198,54 +198,84 @@ public void testCreate() throws Exception {
   @Test
   public void testCreateWithTrust() throws Exception {
     String configsetName = "regular";
-    String configsetSuffix = "testCreateWithTrust";
-    String configsetSuffix2 = "testCreateWithTrust2";
-    uploadConfigSetWithAssertions(configsetName, configsetSuffix, "solr");
-    uploadConfigSetWithAssertions(configsetName, configsetSuffix2, null);
+    String trustedConfigsetSuffix = "testCreateWithTrustTrusted";
+    String untrustedConfigsetSuffix = "testCreateWithTrustUntrusted";
+    uploadConfigSetWithAssertions(configsetName, trustedConfigsetSuffix, "solr");
+    uploadConfigSetWithAssertions(configsetName, untrustedConfigsetSuffix, null);
     try (SolrZkClient zkClient =
         new SolrZkClient(
             cluster.getZkServer().getZkAddress(), AbstractZkTestCase.TIMEOUT, 45000, null)) {
-      assertTrue(isTrusted(zkClient, configsetName, configsetSuffix));
-      assertFalse(isTrusted(zkClient, configsetName, configsetSuffix2));
+      assertTrue(isTrusted(zkClient, configsetName, trustedConfigsetSuffix));
+      assertFalse(isTrusted(zkClient, configsetName, untrustedConfigsetSuffix));
+      // base trusted -> untrusted
+      try {
+        ignoreException("unauthenticated request");
+        // trusted -> untrusted
+        createConfigSet(
+            null, // base is default trusted
+            "abc",
+            Collections.emptyMap(),
+            cluster.getSolrClient(),
+            null); // without username is untrusted
+        fail("Expecting exception");
+      } catch (SolrException e) {
+        assertEquals(SolrException.ErrorCode.UNAUTHORIZED.code, e.code());
+        unIgnoreException("unauthenticated request");
+      }
+      // base trusted -> untrusted
       try {
         ignoreException("unauthenticated request");
-        // trusted -> unstrusted
+        // trusted -> untrusted
         createConfigSet(
-            configsetName + configsetSuffix,
+            "_default", // base is default trusted
+            "def",
+            Collections.emptyMap(),
+            cluster.getSolrClient(),
+            null); // without username is untrusted
+        fail("Expecting exception");
+      } catch (SolrException e) {
+        assertEquals(SolrException.ErrorCode.UNAUTHORIZED.code, e.code());
+        unIgnoreException("unauthenticated request");
+      }
```

Comment:
@epugh these two new tests check the base configset issue that was being run into. Adding these showed the exact same NPE error you were getting. This now returns the correct "unauthorized" error.